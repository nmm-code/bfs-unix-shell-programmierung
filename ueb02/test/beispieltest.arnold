test {
    name    = Beispieltest Teil 1 (stdout)
    comment = Testet die Standardausgabe
    type    = param
    cmd     {cat beispieltest.in | DUT beispieltest.out}
    expect  {^SoRichtigSchlecht$}
}

test {
    name    = Beispieltest Teil 2 (Datei)
    comment = Testet die geschriebene Datei aus Beispieltest Teil 1
    type    = param
    cmd     {diff beispieltest.out beispieltest.exp}
    expect  {^$}
}

test {
    name    = Test 1 Teil 1
    comment = Test auf gleicher Bewertung und sortierung nach Kennzahl
    type    = param
    cmd     {cat test1/test1.in | DUT beispieltest.out}
    expect  {^SoRichtigSchlecht$}
}

test {
     name    = Test 1 Teil 2
    comment = Test auf gleicher Bewertung und sortierung nach Kennzahl
    type    = param
    cmd     {diff beispieltest.out test1/test1.exp}
    expect  {^$}
}

test {
    name    = Test 2 Teil 1
    comment = Failed , groß und kleinschreibung 
    type    = param
    cmd     {cat test2/test2.in | DUT beispieltest.out}
    expect  {^NichtGanzSoSchlecht$}
}

test {
     name    = Test 2 Teil 2
    comment = Failed , groß und kleinschreibung 
    type    = param
    cmd     {diff beispieltest.out test2/test2.exp}
    expect  {^$}
}

test {
    name    = Test 3 Teil 1
    comment = Gleiche Kennzahl und Bewertung
    type    = param
    cmd     {cat test3/test3.in | DUT beispieltest.out}
    expect  {^NichtGanzSoSchlecht$}
}

test {
     name    = Test 3 Teil 2
    comment = Gleiche Kennzahl und Bewertung
    type    = param
    cmd     {diff beispieltest.out test3/test3.exp}
    expect  {^$}
}

test {
    name    = Test 4 Teil 1
    comment = Failed nach einen gültigen Ausdruck
    type    = param
    cmd     {cat test4/test4.in | DUT beispieltest.out}
    expect  {^NochWenigerSchlecht$}
}

test {
     name    = Test 4 Teil 2
    comment = Failed nach einen gültigen Ausdruck
    type    = param
    cmd     {diff beispieltest.out test4/test4.exp}
    expect  {^$}
}

test {
    name    = Test 5 Teil 1
    comment = leere Datei
    type    = param
    cmd     {cat test5/test5.in | DUT beispieltest.out}
    expect  {^$}
}

test {
     name    = Test 5 Teil 2
    comment = leer Datei
    type    = param
    cmd     {diff beispieltest.out test5/test5.exp}
    expect  {^$}
}

test {
    name    = Test 6 Teil 1
    comment = sortierte Datei
    type    = param
    cmd     {cat test6/test6.in | DUT beispieltest.out}
    expect  {^NichtGanzSoSchlecht$}
}

test {
     name    = Test 6 Teil 2
    comment = sortierte Datei
    type    = param
    cmd     {diff beispieltest.out test6/test6.exp}
    expect  {^$}
}