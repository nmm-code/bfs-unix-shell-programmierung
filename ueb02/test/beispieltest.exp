100;NochWenigerSchlecht;3
19;NichtGanzSoSchlecht;1
7;SoRichtigSchlecht;0
