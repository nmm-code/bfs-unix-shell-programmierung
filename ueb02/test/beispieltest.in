19;NichtGanzSoSchlecht;1
100;NochWenigerSchlecht;3
<Failed>
7;SoRichtigSchlecht;0
