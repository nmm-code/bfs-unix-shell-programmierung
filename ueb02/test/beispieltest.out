100;NochWenigerSchlecht;3
99;NochWenigerSchlecht;3
19;NichtGanzSoSchlecht;1
